sql/core/src/main/scala/org/apache/spark/sql/execution/objects.scala

```diff
@@ -68,6 +71,69 @@ case class MapPartitions(
 }
 
 /**
+ * Applies the given function to each input row and encodes the result.
+ *
+ * TODO: Each serializer expression needs the result object which is returned by the given function,
```

Comment:
I would soften this.  Its not clear that removing this small amount of code duplication is worth the confusion that would result in showing the un-common-subexpression-eliminated version of a project in the explain.